src/common/datomish/query/clauses.cljc

```diff
@@ -469,12 +469,15 @@
           bindings (into {} (map (fn [var]
                                    (let [sym (:symbol var)]
                                      [sym [(sql/qualify alias (util/var->sql-var sym))]]))
-                                 free-vars))]
+                                 free-vars))
+
+          known-types
+          (reduce cc/combine-known-types {} (map :known-types ccs))]
```

Comment:
I gave this a shot. Expressed with `transduce`, this is:

```
(transduce (map :known-types) (completing cc/combine-known-types) {} ccs)
```

… which creates an intermediate transducer instead of a lazy sequence, an intermediate function to get a one-argument version of `combine-known-types`, and is slightly longer. Let me know if you prefer that.

(Still, it's nice to know I can figure out `transduce`!)